app/scripts/lib/modal_manager.js

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'jquery',
+  'velocity',
+  'velocityui'
+], function (_, $) {
+  'use strict';
+
+  function ModalManager () {
+    this._viewStates = [];
+    this.$modal = $('#modal');
+
+    // Listen for all clicks
+    $(document).on('click', _.bind(this._closeOnClick, this));
```

Comment:
It looks like we never detach these event handlers, is that the idea behind the `:visible` check in _closeOnClick? "if we're not visible do nothing, else..." It seems like doing those visibility checks on every click will hurt performance a bit, and the code might be simpler if we just fully detach from the DOM on hide. I think we can just mark this as a todo for now and clean up later, though.